pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -687,5 +687,43 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
             arr[k] *= inv_norm;
         }
     }
+
+    /**
+     * @brief Prepares a single computational basis state.
+     *
+     * @param index Index of the target element.
+     */
+    void setBasisState(const std::size_t index) {
+        auto *arr = this->getData();
+        PL_ABORT_IF(index > this->getLength() - 1, "Invalid wire");
+        for (size_t k = 0; k < this->getLength(); k++) {
+            arr[k] = {0.0, 0.0};
+        }
+        arr[index] = {1.0, 0.0};
+    }
+
+    /**
+     * @brief Set values for a batch of elements of the state-vector.
+     *
+     * @param values Values to be set for the target elements.
+     * @param indices Indices of the target elements.
+     */
+    void setStateVector(const std::vector<std::size_t> &indices,
```

Comment:
This was not in the original implementation, but we may want to reset the other state vector entries to `0.0`. Do we?

Reply:
I also wondered if we should check the indices size and compare it with the statevector size.